### ISSUE TITLE
TestDbApps.test_app_resume_tcp_client sometimes blocks indefinitely upon recovery

### ISSUE THREAD

**Issue by aagapi** [actonlang/acton#909]
When running a repeated crash/recovery test on test_app_resume_tcp_client, sometimes the client subprocess spawned by test_db.py remains indefinitely blocked.

Can be verified by repeatedly running:

for i in \$(seq 1000)
do
python3 test_db.py TestDbApps.test_app_resume_tcp_client ; if [ \$? -gt 0 ] ; then break ; fi ; done

(to better observe, increase subprocess timeout for tcp_client in test_db.py to 30 seconds, and remove @flakey from test_app_resume_tcp_client to prevent retries).

All 3 DB servers stay up during this.

Trace:

### Outcome

```
----------------------------------------------------------------------
Ran 1 test in 0.561s

OK
9
:5
??
localhost ??(?????????0??????????????")
?????????
[more garbage output]

======================================================================
ERROR: test_app_resume_tcp_client (__main__.TestDbApps)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_db.py", line 473, in test_app_resume_tcp_client
    self.p = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=30)
  File "/usr/lib/python3.8/subprocess.py", line 495, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "/usr/lib/python3.8/subprocess.py", line 1028, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
  File "/usr/lib/python3.8/subprocess.py", line 1869, in _communicate
    self._check_timeout(endtime, orig_timeout, stdout, stderr)
  File "/usr/lib/python3.8/subprocess.py", line 1072, in _check_timeout
    raise TimeoutExpired(
subprocess.TimeoutExpired: Command '['./rts/ddb_test_client', '47299', '--rts-verbose', '--rts-ddb-replication', '3', '--rts-ddb-host', '127.0.0.1:47100', '--rts-ddb-host', '127.0.0.1:47102', '--rts-ddb-host', '127.0.0.1:47104']' timed out after 30 seconds

----------------------------------------------------------------------
Ran 1 test in 30.571s

FAILED (errors=1)

```

### Expectation

```
TCP client actor should not block indefinitely
```

**Comment by plajjan** [comment]
That garbage output is really quite indicative of DB problems, either in client or server. I guess the garbage bit will be addressed by #916 but the underlying issue remains...

Our challenge here is that the debug output from this script isn't very good, so perhaps the first step is trying to address that.

**Comment by plajjan** [comment]
The test script has been rewritten from scratch and along the way I've fixed a number of actual bugs that I found, one of which may very well have caused this issue. Since so much has changed, I'm inclined to close this ticket as I think it is better to assume it is fixed or that any error would look so different that it should be a new issue either way.

I've run thousands of test runs and while there are bugs, none of been of "hanging" nature, so I really do think this is fixed. Just open new issue for anything else we find! Closing this.
